swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -123,43 +123,22 @@ class Adapter {
         // Tell the UI we're not connected
         self.packetTunnelProvider?.reasserting = true
       }
-
-      lastRelevantPath = nil
     } else {
       if self.packetTunnelProvider?.reasserting == true {
         self.packetTunnelProvider?.reasserting = false
       }
 
-      // Tell connlib to reset network state, but only do so if our connectivity has
-      // meaningfully changed. On darwin, this is needed to send packets
-      // out of a different interface even when 0.0.0.0 is used as the source.
-      // If our primary interface changes, we can be certain the old socket shouldn't be
-      // used anymore.
-      if lastRelevantPath?.connectivityDifferentFrom(path: path) != false {
-        lastRelevantPath = path
-
+      if lastPath?.connectivityDifferentFrom(path: path) != false {
```

Comment:
Thinking about it now, does it make sense to update the path but not react to it? Wouldn't it better to store the last path we performed a reset on and always compare connectivity to that?

Reply:
It's logically the same either way. This way there's one less thing to update inside the conditionally and it's a bit less cognitive overhead IMO.